src/mbgl/tile/geometry_tile.cpp

```diff
@@ -54,6 +54,7 @@ GeometryTile::GeometryTile(const OverscaledTileID& id_,
              parameters.mode,
              parameters.pixelRatio,
              parameters.debugOptions & MapDebugOptions::Collision),
+      fileSource(parameters.fileSource),
```

Comment:
I think it is better to change how `FileSource` is used in renderer. `Map` holds strong ref to filesource and passes `FileSource` reference to renderer, yet, renderer is not owned by map and can outlive it.

Reply:
We should probably change how the `FileSource` works to work the same way as the `BackgroundScheduler`. We pass it everywhere for but in reality it is a singleton, would be better just the class actually using it to retain a reference.

Doesn't need to be addressed by this PR